src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -278,7 +279,7 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
                             {
                                 // Trigger a single load-all operation if a change was detected in one or more key-values with refreshAll: true
                                 applicationSettings = await LoadAll(client, cancellationToken).ConfigureAwait(false);
-
+                                mappedData = await MapConfigurationData(applicationSettings);

```

Comment:
Anytime you call MapConfigurationData, the input dictionary will lose its original ConfigurationSettings because we are not making deep copies before mapping. I think the deep copy logic should move inside MapConfigurationData to avoid any errors.